src/main/java/de/tum/in/www1/artemis/service/AuthorizationCheckService.java

```diff
@@ -27,25 +27,15 @@ public AuthorizationCheckService(UserService userService) {
     }
 
     public <T extends Exercise> boolean isAtLeastTeachingAssistantForExercise(Optional<T> exercise) {
-        if (exercise.isPresent()) {
-            return isAtLeastTeachingAssistantForExercise(exercise.get());
-        }
-        else {
-            return false;
-        }
+        return exercise.filter(this::isAtLeastTeachingAssistantForExercise).isPresent();
```

Comment:
Are you 100% sure that the code does the same? What happens if the exercise is not present?

Reply:
Yes, since `exercise` is an Optional

>If a value is present, apply the provided mapping function to it, and if the result is non-null, return an Optional describing the result. Otherwise return an empty Optional.

So if exercise is not present, the optional will be empty, and therefore `isPresent()` will return false

https://docs.oracle.com/javase/8/docs/api/java/util/Optional.html#filter-java.util.function.Predicate-

Reply:
Put this as comment into the code. Basically this method should have a JavaDoc. Please add one and put these comments into it